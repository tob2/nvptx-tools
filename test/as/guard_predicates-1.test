Verify how 'as' emits guard predicates.


RUN: %target_as_cmd --no-verify %S/guard_predicates-1.s -o %t.o
RUN: cmp %S/guard_predicates-1.o.golden %t.o


..., and, in particular that the 'as' processing is idempotent.

RUN: %target_as_cmd --no-verify %S/guard_predicates-1.o.golden -o %t.o
..., which, currently, it is not.
RUN: cmp %S/guard_predicates-1.o_.golden %t.o
