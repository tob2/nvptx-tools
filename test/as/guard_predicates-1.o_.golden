.func fun
{
.reg .pred p;
.reg .s32 a;
@ p add.s32 a,a,1;
@ !  p add.s32 a,a,2;
@ p add.s32 a,a,3;
@ !  p add.s32 a,a,4;
@ p add.s32 a,a,5;
@ !  p add.s32 a,a,6;
@ p add.s32 a,a,7;
@ !  p add.s32 a,a,8;
@ !  p add.s32 a,a,9;
@ !  p add.s32 a,a,10;
@ p add.s32 a,a,21;
@ !  p add.s32 a,a,22;
@ p add.s32 a,a,31;
@ !  p add.s32 a,a,32;
@ p add.s32 a,a,41;
@ !  p add.s32 a,a,42;
}
